### ISSUE TITLE
Importing "react-loaders" with browserify makes the React DevTools stop working

### ISSUE THREAD

**Issue by Enet4** [jonjaques/react-loaders#2]
I have been thoroughly inspecting my own work looking for a reason why the React DevTools were not showing the components of my web application (the React tab would always show nothing but `<Top Level></Top Level>`). Turns out this module was at fault here.

Here's a reproducible. "index.jsx":

```JavaScript
var React = require('react');
var Loader = require('react-loaders').Loader;

var App = React.createClass({
    render: function(){
        return (
            <div>
               <p>Loading...</p>
            </div>
        )
    }
});

React.render(<App />, document.getElementById('container'));
```

"index.html":

```html
<!DOCTYPE html>
<html>
<head lang="en">
  <meta charset="UTF-8">
  <title>Test Chamber</title>
</head>
<body>
  <div id='container'></div>
  <script src="bundle.js" ></script>
</body>
</html>
```

The build command (also tested an equivalent operation using grunt):

    npm install react reactify react-loaders
    browserify -r react: -t reactify index.jsx -o bundle.js

The outcome is that calling `require` on "react-loaders" alone will render the DevTools' "React" tab useless. I have also heard of other components causing this ([related SO question](http://stackoverflow.com/questions/30229771/making-browserify-bundle-play-nice-with-reactdevtools)).

**Comment by Enet4** [comment]
Apparently, this relates to some problem with webpack configuration file or webpack itself, which is **bundling React alongside the component**! I still haven't managed to fix it though. I'm really starting to think it's a bug with webpack.

**Comment by Enet4** [comment]
All right, I managed to fix the problem with #3. I had to register "react/addons" as an external as well.
![screenshot_2015-05-21_11-03-03](https://cloud.githubusercontent.com/assets/4738426/7746698/67cff838-ffac-11e4-831b-98c6c74e2518.png)

**Comment by jonjaques** [comment]
Hi, thanks for the report! It's probably a bug in my configuration :D  I haven't tried to run this with browserify so I'll check it out tonight and see what I can come up with. Thanks for the submissions so far.

**Comment by Enet4** [comment]
It was also my pleasure to have this pesky matter sorted out. I even learned a bit more about webpack on the way. ;)

**Comment by jonjaques** [comment]
Sorry for the wait. Been busy the past week with new job :D 

This should be fixed with v1.2.3. Note there are in-page requirements for the React Devtools (https://github.com/facebook/react-devtools#the-react-tab-doesnt-show-up), also it seems to only work when running from an http-server.

Browserify doesn't seem to be able to handle transforming `require('react')` to global `React` references without browserify-shim, which is okay. It's a bit easier in webpack, you can just specify `external: { react: 'React' }`, and then load it from CDN.

**Comment by Enet4** [comment]
> This should be fixed with v1.2.3.

Good to know!

> Note there are in-page requirements for the React Devtools (https://github.com/facebook/react-devtools#the-react-tab-doesnt-show-up)

I have read that over and over again when trying to figure the problem out. Besides, the React tab _was_ showing up, it just din't show the expected content, which is a seldom mentioned symptom.

>, also it seems to only work when running from an http-server.

I actually managed to make it work local. One just has to allow file URL access in the extension.

> Browserify doesn't seem to be able to handle transforming require('react') to global React
> references without browserify-shim, which is okay. Instead, I'm creating a separate 'react' bundle,
> and then specifying it as 'external', which allows it to be require()'d in the global context. It's a bit
> easier in webpack, you can just specify external: { react: 'React' }, and then load it from CDN.

The most frequent use case would be the web application bundling React alongside all other modules, including components. With this approach, react components are expected to `require` React. Nevertheless, that may work as well.

**Comment by jonjaques** [comment]
Yup, the basic requirement is the devtools need to be able to detect React in the global environment. Whether that be through `require()` or `window.React`. Webpack and browserify's default is to wrap everything up in a closure, so you have to go with either the shim or the solution outlined in `npm test`.

I also found that without specifying the `displayName` property, I couldn't get past the `<TopLevel />` either, so it may have something to do with that.
